prow/plugins/cherrypickapproved/cherrypickapproved.go

```diff
@@ -0,0 +1,228 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cherrypickapproved
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/labels"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	// PluginName defines this plugin's registered name.
+	PluginName = labels.CpApproved
+
+	defaultKubernetesRepoAndOrg = "kubernetes"
+)
+
+var (
+	branchRegexp = regexp.MustCompile("^release-*")
```

Comment:
This should be configurable via flag or config file (whichever is possible for built-in plugin). This plugin might also be used by projects other than Kubernetes, and those projects might use some different branch regexp (e.g. `release/v*`)

Reply:
Done :+1: